enterprise/internal/batches/background/batch_spec_resolution_worker.go

```diff
@@ -0,0 +1,75 @@
+package background
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/keegancsmith/sqlf"
+
+	"github.com/sourcegraph/sourcegraph/enterprise/internal/batches/store"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/internal/workerutil"
+	"github.com/sourcegraph/sourcegraph/internal/workerutil/dbworker"
+	dbworkerstore "github.com/sourcegraph/sourcegraph/internal/workerutil/dbworker/store"
+)
+
+const batchSpecResolutionMaxNumRetries = 60
```

Comment:
```suggestion
const batchSpecResolutionMaxNumRetries = 0
```

I don't think we want retries here. If it's ephemeral it might be nice, but often it'll just be invalid user input, I think